httplib.h

```diff
@@ -2521,12 +2519,12 @@ inline int shutdown_socket(socket_t sock) {
 template <typename BindOrConnect>
 socket_t create_socket(const char *host, const char *ip, int port,
                        int address_family, int socket_flags, bool tcp_nodelay,
-                       SocketOptions socket_options,
+                       const SocketOptions &socket_options,
                        BindOrConnect bind_or_connect) {
   // Get address info
   const char *node = nullptr;
-  struct addrinfo hints;
-  struct addrinfo *result;
+  struct addrinfo hints {};
```

Comment:
Initializing `hints` here is useless, as `std::memset` is called below to zero-initialize the struct as well. You either initialize it with `{}` or `memset`